rviz_default_plugins/include/rviz_default_plugins/displays/interactive_markers/interactive_marker_control.hpp

```diff
@@ -0,0 +1,566 @@
+
+  /**
+   * PointsMarkers are rendered by special shader programs, so the
+   * regular highlighting method does not work for them.
+   * Keep a vector of them so we can call their setHighlightColor() function.
+   */
+  std::vector<markers::PointsMarker::SharedPtr> points_markers_;
+
+  /// Stores the rotation around the x axis of the control.
+  /**
+   * Only relevant for fixed-orientation rotation controls.
+  */
+  Ogre::Radian rotation_;
+
+  /// Stores the rotation around the x axis of the control when the mouse-down event happened.
+  /**
+   * Only relevant for fixed-orientation rotation controls.
+  */
+  Ogre::Radian rotation_at_mouse_down_;
+
+  /**
+   * The 3D position of the mouse click/cursor when the 'grab' button is
+   * pressed, relative to the reference frame.
+  */
+  Ogre::Vector3 grab_point_in_reference_frame_;
+
+  /**
+   * The orientation of the cursor when the 'grab' button is pressed, relative
+   * to the reference frame.
+  */
+  Ogre::Quaternion grab_orientation_in_reference_frame_;
+
+  /**
+   * Records the 3D position of the cursor relative to the parent marker,
+   * expressed in the cursor frame, when the 'grab' button is pressed.
+   */
+  Ogre::Vector3 parent_to_cursor_in_cursor_frame_at_grab_;
+
+  /// Records the rotation of the parent from the cursor frame when the 'grab' button is pressed.
+  Ogre::Quaternion rotation_cursor_to_parent_at_grab_;
+
+  /// The modifier state when drag begins.
+  Qt::KeyboardModifiers modifiers_at_drag_begin_;
+
+  /// X-position of the mouse when drag begins.
+  int mouse_x_at_drag_begin_;
+
+  /// Y-position of the mouse when drag begins.
+  int mouse_y_at_drag_begin_;
+
+  /// Mouse ray when drag begins.
+  Ogre::Ray mouse_ray_at_drag_begin_;
+
+  /// How far to move in Z when the mouse moves 1 pixel.
+  double mouse_z_scale_;
+
+  /// X-offset of the absolute mouse position from the relative mouse position.
+  int mouse_relative_to_absolute_x_;
+
+  /// Y-offset of the absolute mouse position from the relative mouse position.
+  int mouse_relative_to_absolute_y_;
+
+  /// Position of grab relative to parent in world coordinates.
+  // Ogre::Vector3 parent_to_grab_position_; // obsolete now, but left for ABI compatibility
+
+  /// The position of the parent when the mouse button is pressed.
+  Ogre::Vector3 parent_position_at_mouse_down_;
+
+  /// The orientation of the control_frame_node_ when the mouse button is pressed.
+  Ogre::Quaternion control_frame_orientation_at_mouse_down_;
+
+  /// The orientation of the parent when the mouse button is pressed.
+  Ogre::Quaternion parent_orientation_at_mouse_down_;
+
+  /**
+   * The direction vector of the axis of rotation during a mouse
+   * drag, relative to the reference frame.
+   * Computed on mouse down event.
+  */
+  Ogre::Vector3 rotation_axis_;
+
+  /// The center of rotation during a mouse drag, relative to the control frame.
+  /**
+   * Computed on mouse down event.
+  */
```

Comment:
nitpick: alignment of `*` is wrong here.

Reply:
5dfa654